src/test/rustdoc/const-generics/const-generics-docs.rs

```diff
@@ -0,0 +1,128 @@
+// edition:2018
+// aux-build: extern_crate.rs
+#![feature(min_const_generics)]
+#![crate_name = "foo"]
+
+extern crate extern_crate;
+// @has foo/fn.extern_fn.html '//pre[@class="rust fn"]' \
+//      'pub fn extern_fn<const N: usize>() -> impl Iterator<Item = [u8; N]>'
+pub use extern_crate::extern_fn;
+// @has foo/struct.ExternTy.html '//pre[@class="rust struct"]' \
+//      'pub struct ExternTy<const N: usize> {'
+pub use extern_crate::ExternTy;
+// @has foo/type.TyAlias.html '//pre[@class="rust typedef"]' \
+//      'type TyAlias<const N: usize> = ExternTy<N>;'
+pub use extern_crate::TyAlias;
+// @has foo/trait.WTrait.html '//pre[@class="rust trait"]' \
+//      'pub trait WTrait<const N: usize, const M: usize>'
+// @has - '//*[@class="rust trait"]' 'fn hey<const P: usize>() -> usize'
+pub use extern_crate::WTrait;
+
+// @has foo/trait.Trait.html '//pre[@class="rust trait"]' \
+//      'pub trait Trait<const N: usize>'
+// @has - '//*[@id="impl-Trait%3C1_usize%3E-for-u8"]//code' 'impl Trait<1_usize> for u8'
+// @has - '//*[@id="impl-Trait%3C2_usize%3E-for-u8"]//code' 'impl Trait<2_usize> for u8'
```

Comment:
Is it intentional that this adds a `_usize` suffix? I'm not opposed but it shouldn't be by accident.

Reply:
not sure, it's the way `Display` for `ty::Const` is currently implemented.

It looks like rustdoc already explicitly truncates `_usize` for array lengths in `fn print_const`, so I think that this is probably at least somewhat intentional

Reply:
Hmm, ok - do you mind extending that truncation to generic const arguments too?

Alternatively I'd love for this to be changed in ty::Const so rustdoc doesn't have to keep working around it.

Reply:
I meant that we intentionally add the type here as it isn't obvious otherwise.
I interpret the comment in `print_const` to mean that we want these type annotations outside of arrays. 

> // array lengths are obviously usize

I personally don't have a preference here, one the one hand it is a bit verbose, but it can be helpful to read `Foo<3_u8, 16_usize>` instead of just `Foo<3, 16>`. isn't really relevant to this PR though, so I opened a topic on zulip https://rust-lang.zulipchat.com/#narrow/stream/260443-project-const-generics/topic/documenting.20concrete.20consts.20in.20types

Reply:
Yeah, it definitely doesn't need to block new tests.